modules/atomic-widgets/styles/style-schema.php

```diff
@@ -0,0 +1,81 @@
+<?php
+namespace Elementor\Modules\AtomicWidgets\Styles;
+
+use Elementor\Modules\AtomicWidgets\PropTypes\Linked_Dimensions_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\Number_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\Size_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\String_Prop_Type;
+
+class Style_Schema {
+	public static function get() {
+		return array_merge(
+			self::get_size_props(),
+			self::get_position_props(),
+			self::get_typography_props(),
+			self::get_spacing_props()
+		);
+	}
+
+	private static function get_size_props() {
+		return [
+			'width' => Size_Prop_Type::make(),
+			'height' => Size_Prop_Type::make(),
+			'min-width' => Size_Prop_Type::make(),
+			'min-height' => Size_Prop_Type::make(),
+			'max-width' => Size_Prop_Type::make(),
+			'max-height' => Size_Prop_Type::make(),
+		];
+	}
+
+	private static function get_position_props() {
+		return [
+			'z-index' => Number_Prop_Type::make(),
+		];
+	}
+
+	private static function get_typography_props() {
+		return [
+			'font-weight' => Number_Prop_Type::make()->enum([
+				100,
+				200,
+				300,
+				400,
+				500,
+				600,
+				700,
+				800,
+				900,
+			]),
+			'font-size' => Size_Prop_Type::make(),
+			'color' => String_Prop_Type::make(),
+			'letter-spacing' => Size_Prop_Type::make(),
+			'word-spacing' => Size_Prop_Type::make(),
+			'text-align' => String_Prop_Type::make()->enum([
+				'left',
+				'center',
+				'right',
+				'justify',
+			]),
+			'font-style' => String_Prop_Type::make()->enum([
+				'normal',
+				'italic',
+				'oblique',
+			]),
+			// TODO: validate text-decoration in more specific way [EDS-524]
+			'text-decoration' => String_Prop_Type::make(),
```

Comment:
For now it is just string right? `underline`, `wavy`? maybe for now it will be `enum`?

Reply:
It's saved as a string and not as an array, for example: 'underline overline'